src/controllers/followController.js

```diff
@@ -0,0 +1,167 @@
+import { config } from 'dotenv';
+import { User, Follow } from '../sequelize/models';
+
+config();
+/**
+ * @description author follow each other
+ */
+class Following {
+  /**
+   *
+   * @param {*} req
+   * @param {*} res
+   * @returns {object} return object containg data
+   */
+  static async followUser(req, res) {
+    const { userName } = req.params;
+    const userId = req.userData.id;
+
+    const checkUser = await User.findOne({ where: { userName } });
+    const { id } = checkUser;
+    const checkFollower = await Follow.findOne({ where: { follower: userId, following: id } });
+
+    if (!checkUser) {
+      return res.status(404).json({
+        error: `User name ${userName} not exists`
+      });
+    }
+
+    if (checkFollower) {
+      return res.status(409).json({
+        error: `User name ${userName} already followed`
+      });
+    }
+
+    if (checkUser.id === userId) {
+      return res.status(403).json({
+        error: 'You can not follow yourself'
+      });
+    }
+
+    const newFollowing = {
+      follower: userId,
+      following: checkUser.id
+    };
+    const newSavedFollow = await Follow.create(newFollowing);
+    return res.status(201).json({
+      data: newSavedFollow
+    });
+  }
+
+  /**
+   *
+   * @param {*} req
+   * @param {*} res
+   * @returns {object} return object containg message
+   */
+  static async unFollowUser(req, res) {
+    const { userName } = req.params;
+    const userId = req.userData.id;
+
+    const getUserId = await User.findOne({ where: { userName } });
+
+    if (!getUserId) {
+      return res.status(404).json({
+        error: `User name ${userName} not exists`
+      });
+    }
+
+    const { id } = getUserId;
+    const checkFollower = await Follow.findOne({ where: { follower: userId, following: id } });
+
+    if (!checkFollower) {
+      return res.status(409).json({
+        error: `User name ${userName} not followed by you`
+      });
+    }
+
+    await Follow.destroy({ where: { follower: userId, following: id } });
+    return res.status(200).json({
+      message: `User ${userName} removed in your following`
+    });
+  }
+
+  /**
+   *
+   * @param {*} req
+   * @param {*} res
+   * @returns {object} return object containg data
+   *
+   */
+  static async getFollowingList(req, res) {
+    const userList = [];
+    const { userName } = req.params;
+
+    const getUserId = await User.findOne({ where: { userName } });
+
+    if (!getUserId) {
+      return res.status(404).json({
+        error: `User name ${userName} not exists`
+      });
+    }
+
+    const userId = getUserId.id;
+    const followingList = await Follow.findAll({ where: { follower: userId } });
+
+    if (followingList.length === 0) {
+      return res.status(404).json({
+        message: 'No followers'
```

Comment:
this message is misinforming

Reply:
well, i will find the best message

Reply:
Instead of **No followers**, You can use this one: **You currently have no followers**. And in case, you have not followed anyone, You can use: **You are currently not following anyone**

Reply:
Thank you @placideirandora

Reply:
Done!